beacon-chain/rpc/prysm/v1alpha1/validator/proposer_attestations.go

```diff
@@ -287,3 +295,211 @@ func (vs *Server) deleteAttsInPool(ctx context.Context, atts []ethpb.Att) error
 	}
 	return nil
 }
+
+// isAttestationFromCurrentEpoch returns true if the attestation is from the current epoch.
+func (vs *Server) isAttestationFromCurrentEpoch(att ethpb.Att) bool {
+	currentSlot := vs.TimeFetcher.CurrentSlot()
+	currentEpoch := slots.ToEpoch(currentSlot)
+	attestationSlot := att.GetData().Slot
+	attestationEpoch := slots.ToEpoch(attestationSlot)
+	return attestationEpoch == currentEpoch
+}
+
+// isAttestationFromPreviousEpoch returns true if the attestation is from the previous epoch.
+func (vs *Server) isAttestationFromPreviousEpoch(att ethpb.Att) bool {
+	currentSlot := vs.TimeFetcher.CurrentSlot()
+	previousEpoch := slots.ToEpoch(currentSlot)
+	if previousEpoch >= 1 {
+		previousEpoch = previousEpoch.Sub(1)
+	}
+	attestationSlot := att.GetData().Slot
+	attestationEpoch := slots.ToEpoch(attestationSlot)
+	return attestationEpoch == previousEpoch
+}
+
+// filterCurrentEpochAttestationByForkchoice filters attestations from the current epoch based on fork choice conditions.
+// Returns true if any of the following conditions are met:
+// 1. The attestation beacon block root is for a slot in the previous epoch (according to fork choice).
+// 2. The attestation target root is the same as the attestation beacon block root.
+// 3. The common ancestor of the head block and the attestation beacon block root is from the previous epoch.
+func (vs *Server) filterCurrentEpochAttestationByForkchoice(ctx context.Context, att ethpb.Att, headRoot [32]byte) (bool, error) {
+	if !vs.isAttestationFromCurrentEpoch(att) {
+		return false, nil
+	}
+
+	attTargetRoot := [32]byte(att.GetData().Target.Root)
+	attBlockRoot := [32]byte(att.GetData().BeaconBlockRoot)
+	if attBlockRoot == attTargetRoot {
```

Comment:
maybe i am missing something, why would this be valid ? especially if they have a different target to ours. It just means the attestations at slot 0 is consistent.